### PR TITLE
[talib] New port

### PR DESCRIPTION
**Pull request by kkqy**
This PR resumes work on https://github.com/microsoft/vcpkg/issues/29328

- [ ] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [ ] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [ ] The versioning scheme in `vcpkg.json` matches what upstream says.
- [ ] The license declaration in `vcpkg.json` matches what upstream says.
- [ ] The installed as the "copyright" file matches what upstream says.
- [ ] The source code of the component installed comes from an authoritative source.
- [ ] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [ ] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [ ] Only one version is in the new port's versions file.
- [ ] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by kkqy** [comment]
@microsoft-github-policy-service agree

File: ports/talib/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,85 @@
+set(7ZIP_VERSION "0.4.0")
```
Comment (dg0yt) [inline ports/talib/portfile.cmake:1]: Just use `${VERSION}`, it is taken from `vcpkg.json`.

File: ports/talib/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+vcpkg_download_distfile(
+    ARCHIVE
+    URLS "https://jaist.dl.sourceforge.net/project/ta-lib/ta-lib/${7ZIP_VERSION}/ta-lib-${7ZIP_VERSION}-msvc.zip"
+    FILENAME "ta-lib-${7ZIP_VERSION}-msvc.zip"
+    SHA512 5f211327b6a1d4f00d0a2b9e276adadd118d7aa29fc87c6771d550fda124a863b4a20e3803f325f7c903c82ea12bfb23121a5f0566eeaa434e0f107a6eedb737
+)
```
Comment (dg0yt) [inline ports/talib/portfile.cmake:8]: AFAIU thus URL is just one mirror. Can't the port use
https://learn.microsoft.com/en-us/vcpkg/maintainers/functions/vcpkg_from_sourceforge ?

**Review by dg0yt — COMMENTED** [review]
Can't this port use https://learn.microsoft.com/en-us/vcpkg/maintainers/functions/vcpkg_build_nmake?

**Comment by kkqy** [comment]
> Can't this port use https://learn.microsoft.com/en-us/vcpkg/maintainers/functions/vcpkg_build_nmake?

@dg0yt 
I am sorry , I consults the json file of 7z,  but I forgot to modify the variable's name.
because of the problem of working directory and relative path ,so I use vcpkg_execute_build_process before.
but I find that the "PROJECT_SUBPATH" parameter of vcpkg_build_nmake can change the working directory,
and it looks work well, so I use vcpkg_build_nmake and commit again.

**Comment by kkqy** [comment]
because the talib's makefiles of debug and release are individual, we can't use vcpkg_build_nmake directly.

```
These directories contains the Makefiles
for building TA-Lib.

The first level of directory allows to identify
the kind of application environment in which the
static library is going to be used:

  CSR - Single thread
  CMR - Multi  thread
  CSD - Single thread with debug options
  CMD - Multi  thread with debug options
  CSP - Single thread with profiling options
  CMP - Multi  thread with profiling options
  CDR - DLL Release (Win32 only)
  CDD - DLL Debug (Win32 only)

Choose the most appropriate directory depending of
your platform/compiler and run "make" or "nmake"
from there.

See the API Documentation for more information
on how to build and link with TA-Lib.

The directory gen_make contains the perl scripts
automating the generation of all the makefiles.

```

**Comment by kkqy** [comment]
Why does the workflow microsoft.vcpkg.pr (x86_windows) show "versions/t-/talib.json(2,): error : while parsing versions for talib from version/t-/talib.json",
Is there no detail?

**Review by dg0yt — COMMENTED** [review]
> because the talib's makefiles of debug and release are individual, we can't use vcpkg_build_nmake directly.

I would normally solve this by adding a custom top-level makefile. It could also handle the install step - a single rule which is invoked for each build type.

But I wonder if we sould package 0.4.0 at all (from 2007!),
Later commits add a `CMakeLists.txt`.

There are no new releases on SF but GH has a [v0.6.0-rc.1](https://github.com/TA-Lib/ta-lib/releases/tag/v0.6.0-rc.1) tag (from 2016). And this year's commit say GH becomes the official repo.
https://github.com/TA-Lib/ta-lib/blob/main/README.md

**Comment by kkqy** [comment]
> > because the talib's makefiles of debug and release are individual, we can't use vcpkg_build_nmake directly.
> 
> I would normally solve this by adding a custom top-level makefile. It could also handle the install step - a single rule which is invoked for each build type.
> 
> But I wonder if we sould package 0.4.0 at all (from 2007!), Later commits add a `CMakeLists.txt`.
> 
> There are no new releases on SF but GH has a [v0.6.0-rc.1](https://github.com/TA-Lib/ta-lib/releases/tag/v0.6.0-rc.1) tag (from 2016). And this year's commit say GH becomes the official repo. https://github.com/TA-Lib/ta-lib/blob/main/README.md

0.4.0 is the most popular version, its python package [ta-lib-python](https://github.com/TA-Lib/ta-lib-python) which is more popular than it self is based on this version 0.4.0.
so I think 0.4.0 should be stable and reliable.

**Comment by Cheney-W** [comment]
> Why does the workflow microsoft.vcpkg.pr (x86_windows) show "versions/t-/talib.json(2,): error : while parsing versions for talib from version/t-/talib.json", Is there no detail?

You could do below command:
```
vcpkg x-add-version talib --overwrite-version
git add .
git commit --amend --no-edit
git push
```

**Comment by kkqy** [comment]
I combined all changes.

**Review by Cheney-W — DISMISSED** [review]

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
> 0.4.0 is the most popular version, its python package [ta-lib-python](https://github.com/TA-Lib/ta-lib-python) which is more popular than it self is based on this version 0.4.0.
> so I think 0.4.0 should be stable and reliable.

I think it is acceptable to publish a version that people are mostly using. However I agree with @dg0yt 's observation here:

> There are no new releases on SF but GH has a [v0.6.0-rc.1](https://github.com/TA-Lib/ta-lib/releases/tag/v0.6.0-rc.1) tag (from 2016). And this year's commit say GH becomes the official repo.
> https://github.com/TA-Lib/ta-lib/blob/main/README.md
 
The version we are supposed to offer by default is whatever upstream says it is. If upstream wants to move to github and has a 0.6.0, that should be added shortly after this PR lands if possible.

**Pull request merged**